perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_adName.java

```diff
@@ -34,15 +35,26 @@
 
 	private static final String A_R_D_AD_OU_NAME = AttributesManager.NS_RESOURCE_ATTR_DEF + ":adOuName";
 
+	private static final Pattern pattern = Pattern.compile("(\\d|\\w|-|\\.)*");
+
 	@Override
 	public void checkAttributeValue(PerunSessionImpl sess, Resource resource, Group group, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
-		String attributeValue = null;
+		String attributeValue;
 
 		//Attribute can be null
 		if(attribute.getValue() == null) {
 			return;
 		}
 
+		if(attribute.getValue() instanceof String) {
+			attributeValue = (String) attribute.getValue();
+			if (!pattern.matcher(attributeValue).matches()) {
+				throw new WrongAttributeValueException(attribute, "Invalid attribute adName value. It should contain only letters, digits, hyphens or dots.");
```

Comment:
Please mention also underscore as allowed character, since its allowed by ``\\w`` in a regex and its correct by redmine task description.

Also, isn't regex too complex ? @licehammer I believe ``\\w`` contains ``\\d`` - https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html so we can use only ``\\w`` ?

Reply:
@zlamalp 
I will mention the underscore. And I will siplify the regex, I thought the \\w corresponds to [a-z_A-Z].
My bad.